trollsift/parser.py

```diff
@@ -390,7 +391,7 @@ def _convert(convdef, stri):
     is_fixed_point = any([ftype in convdef for ftype in fixed_point_types])
     if '%' in convdef:
         result = dt.datetime.strptime(stri, convdef)
-    elif 'd' in convdef or 's' in convdef or is_fixed_point:
+    elif 'd' in convdef or 's' in convdef or '' in convdef or is_fixed_point:
```

Comment:
Is `convdef` a string? If so, then this addition is always true:

```python
In [1]: "" in "abc"
Out[1]: True
```

Reply:
Indeed. I thought, for whatever reason, that that was an iterable, but it is indeed a string, so that is not OK.

Would it be OK to have `result = _strip_padding(convdef, stri)` in line 404 instead of just `stri`? This passes the tests, but I want to check with you first.

Reply:
I think so? The fact that there is this `if` statement makes me a little concerned that we'll be screwing something up, but I'm not sure what right now. I suppose you could restructure this whole block of code if you did this with something like:

```python
if '%' in convdef:
    ...
else:
    result = _strip_padding(convdef, stri)
    if 'd' in convdef:
        result = int(result)
    ...
```

Reply:
Done.